### ISSUE TITLE
Using Memcode at schools

### ISSUE THREAD

**Issue by lakesare** [lakesare/memcode#126]
To enable Memcode usage at schools:
- [x] We should implement email signups
- [x] We should implement student statistics visible to the course author (quote: "moreover, it would be insane to have a little interface showing us (so: teachers / authors) how many people (so: pupils) reviewed the deck, the percentage of right answers, etc")
- [x] We should implement the genially iframe embeds with good styling

Email exchange (inaccessible to anyone but the owner): https://mail.google.com/mail/u/1/#inbox/jrjtXLCdDTVGSHwNdvPfVgSdFNRNdfjRKdfSXSxwgHDDVpSLFRTMCGbjsRJjVMgklmmWLsfh.

**Comment by kgashok** [comment]
@lakesare 
Want to fully understand what the motivations are around this issue. Probably worth moving to Github discussions?

**Comment by lakesare** [comment]
There's a teacher in France that uses Memcode with his pupils (see the [video](https://www.youtube.com/watch?v=2L-eqaVDDJo&t=773s)), and he needed a couple of updates which would enable Memcode to be used in schools.

1. Email signups are needed because students frequently have gmail blocked on their school laptops.
2. Student statistics are necessary for a teacher.
3. Genially is a presentation site (like powerpoint), and I implemented this feature already.

**Comment by piotrvidal** [comment]
This is a genius idea. I was studying earlier today and thought it'd be cool to see some stats and that's why I came here to see if such thing was being worked on :)

Count me in to help out with the implementation of some .js chart library.

**Comment by lakesare** [comment]
@piotrvidal, agreed, I'd enjoy seeing some stats too.

For example, "43 students" could be clickable, and it could display:
- profile picture of a student (with a link to their profile)
- last time they reviewed the course
- number of flashcards they have learned, etc.

<img width="169" alt="image" src="https://user-images.githubusercontent.com/7578559/152616629-a83aac92-f8a5-4898-90a5-e455c6da3b12.png">

As always - welcome to PR!

**Comment by piotrvidal** [comment]
Hello,

Where's some example env.js I can use ? (swapping login vars with my own
local db's)

**Comment by lakesare** [comment]
@piotrvidal, please write directly to **contact@memcode.com**, or tell me your email here - I can't send you the env file, because Github won't give me your email.

**Comment by piotrvidal** [comment]
Lol yeah I just figured that out :smile: you just got an email :)

**Comment by piotrvidal** [comment]
@lakesare I've made the "X students" label clickable and redirect to:
`/users/${this.props.courseDto.author.id}/${this.props.courseDto.course.id}/students-stats`

I'll create the view with the profile pics of students and their stats after my exam session finishes.

There are two things I wanted to ask, though:

- What could be causing that I need to `make frontend-webpack` and `make start` after every change to see them? (I work with Angular, not React, and hot reload always worked out of the box with angular-cli, I'll also google this stuff after the exams too but maybe you already know what's up)
- Should we create a specific task for this and have it assigned to me? This task about schools is a two-thing task (email logins + view with stats)

**Comment by lakesare** [comment]
I'm not sure about a separate page for the stats, how about a modal (see '~/components/TogglerAndModal')?

- You don't need to run `make frontend-webpack` and `make start` repeatedly, they get updated on their own! You only need to refresh the browser window with CMD+R.
- Agreed, I'll create the new task. Let's also invite you to our Slack (I sent you an email), I'll be able to answer more quickly there.

**Comment by sb0001** [comment]
As a teacher I thank you for memcode!

Features I'd like to see VERY much:
- email signups - it doesn't make sense to let all students create github accounts... gmail is not really an option
- for language-courses case sensitivity would be important (english vs English) - maybe as an option per course?
- to improve privacy a user should be able to hide her/his statistics from the public. It makes sense, to show the statistics to the course-owner or to the user's teacher (voluntarily) in case you want teachers to re-use others' courses or to show the statistics just to the members of a certain group/a class, once such a feature has been developed.

Features that would be helpful:
- being able to organise courses:
  - first year
    - unit 1
    - unit 2
    - ...
  - second year
    - unit 1
    - unit 2

- being able to organise students (and see their progress):
   - class 1a
     - student 1
     - student 2
     - ...
  - class 1b
    - ...

**Comment by lakesare** [comment]
### Price assignment (re: [memcode#requesting-a-feature](https://github.com/lakesare/memcode#requesting-a-feature))

**Enable email signups**: **$100**
**Case-sensitivity option**: **$30**
**Hiding statistics from the public option**: **$30**
**Student organization interface**: **$200**
